src/shared/hooks/use-prevent-back-navigation.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+const usePreventBackNavigation = (redirectUrl?: string) => {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const handlePopState = () => {
+      window.history.pushState(null, '', window.location.href);
+
+      if (redirectUrl) {
+        navigate(redirectUrl, { replace: true });
+      }
+    };
```

Comment:
```suggestion
 ```suggestion
    const handlePopState = () => {
      if (redirectUrl) {
        navigate(redirectUrl, { replace: true });
      } else {
        window.history.pushState(null, '', window.location.href);
      }
    };
```
```

만약에 redirectUrl이 먼저 지정되어있다면 그냥 거기로 먼저 이동시키면 되는거 아닌가 해서.. 먼저 다시 히스토리 스택에 pushState로 현재 주소를 또 추가해야하나 궁금해서 여쭤봅니다 !! 

흐름이 잘 안잡혀서요 ㅠㅠ

Reply:
원래 뒤로가기 '막기' 자체에 집중했는데, 희선 님 리뷰 보고 생각해 보니까, 저희는 뒤로가기를 막는 것보다 원하는 페이지로 이동시키는 게 중요하다고 생각해서

```ts
import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';

const usePreventBackNavigation = (redirectUrl: string) => {
  const navigate = useNavigate();

  useEffect(() => {
    const handlePopState = () => {
      navigate(redirectUrl, { replace: true });
    };

    window.history.pushState(null, '', window.location.href);
    window.addEventListener('popstate', handlePopState);

    return () => {
      window.removeEventListener('popstate', handlePopState);
    };
  }, [navigate, redirectUrl]);
};

export default usePreventBackNavigation;
```
이렇게 간단하게 수정했습니다!

Reply:
좋습니다 👍